desc/vmec_utils.py

```diff
@@ -468,3 +482,451 @@ def vmec_boundary_subspace(eq, RBC=None, ZBS=None, RBS=None, ZBC=None):  # noqa:
+
+    timer.stop("Boozer Transform")
+    if verbose > 1:
+        timer.disp("Boozer Transform")
+
+    file = Dataset(path, mode="w", format="NETCDF3_64BIT_OFFSET")
+    # dimensions
+    # a few of these are redundant, but are included for sake
+    # of matching the convention of the original booz_xform outputs
+    # and the hidden symmetries implementation:
+    # (below two lines are a single link)
+    # https://github.com/hiddenSymmetries/booz_xform/blob/main/src/...
+    # _booz_xform/write_boozmn.cpp
+    file.createDimension("radius", s_full.size)  # number of flux surfaces plus 1
+    file.createDimension("comput_surfs", surfs - 1)  # number of flux surfaces
+    file.createDimension("pack_rad", surfs)  # number of flux surfaces
+    file.createDimension("mn_mode", num_modes)  # number of Fourier modes
+    file.createDimension("mn_modes", num_modes)  # number of Fourier modes
+    file.createDimension("preset", 21)  # dimension of profile inputs
+    file.createDimension("ndfmax", 101)  # used for am_aux & ai_aux
+    file.createDimension("time", 100)  # used for fsq* & wdot
+    file.createDimension("dim_00001", 1)
+    file.createDimension("dim_00020", 20)
+    file.createDimension("dim_00100", 100)
+    file.createDimension("dim_00200", 200)
+
+    version_ = file.createVariable("version", "S1", ("dim_00100",))
+    version_str = "DESC Python implementation of booz_xform"
+    version_[:] = stringtochar(
+        np.array(
+            [" " * (100 - len(version_str))],
+            "S" + str(file.dimensions["dim_00100"].size),
+        )
+    )
+
+    nfp = file.createVariable("nfp_b", np.int32)
+    nfp.long_name = "number of field periods"
+    nfp[:] = NFP
+
+    lasym = file.createVariable("lasym__logical__", np.int32)
+    lasym.long_name = "0 if the configuration is stellarator-symmetric, 1 if not"
+    lasym[:] = not eq.sym
+
+    ns = file.createVariable("ns_b", np.int32)
+    ns.long_name = "Number of radial surfaces at which data is outputted minus 1"
+    ns[:] = surfs
+
+    aspect = file.createVariable("aspect_b", np.float64)
+    aspect.long_name = "Aspect Ratio"
+    aspect[:] = eq.compute("R0/a")["R0/a"]
+
+    grid_lcfs = LinearGrid(M=eq.M_grid, N=eq.N_grid, rho=1, NFP=NFP)
+    Rs = eq.compute("R", grid=grid_lcfs)["R"]
+    Rmax = file.createVariable("rmax_b", np.int32)
+    Rmax.long_name = "Maximum Radius"
+    Rmax[:] = np.max(Rs)
+
+    Rmin = file.createVariable("rmin_b", np.int32)
+    Rmin.long_name = "Minimum Radius"
+    Rmin[:] = np.min(Rs)
+
+    # betaxis = beta_vol at the axis?
+    betaxis = file.createVariable("betaxis_b", np.float64)
+    betaxis[:] = 0
+    betaxis.long_name = "Not Implemented: This output is hard-coded to 0!"
+
+    mboz = file.createVariable("mboz_b", np.int32)
+    mboz.long_name = (
+        "Maximum poloidal mode number m for which the Fourier"
+        "amplitudes rmnc, bmnc etc are stored"
+    )
+    mboz[:] = M_booz
```

Comment:
shouldn't this be -1 or something?

Reply:
This currently matches the C++ version's value when the same arguments for mboz and nboz are passed in to their function as our function.